airflow/sensors/external_task_sensor.py

```diff
@@ -68,23 +70,35 @@ def __init__(self,
                  external_dag_id,
                  external_task_id=None,
                  allowed_states=None,
+                 failed_states=None,
                  execution_delta=None,
                  execution_date_fn=None,
                  check_existence=False,
                  *args,
                  **kwargs):
         super().__init__(*args, **kwargs)
         self.allowed_states = allowed_states or [State.SUCCESS]
+        self.failed_states = failed_states or []
+
+        total_states = []
+        total_states.extend(self.allowed_states)
+        total_states.extend(self.failed_states)
+
+        if len(list(set(total_states))) < (len(self.failed_states) + len(self.allowed_states)):
+            raise AirflowException("Duplicate values provided as allowed "
+                                   "`{}` and failed states `{}`"
+                                   .format(self.allowed_states, self.failed_states))
+
         if external_task_id:
-            if not set(self.allowed_states) <= set(State.task_states):
+            if not set(total_states) <= set(State.task_states):
                 raise ValueError(
-                    'Valid values for `allowed_states` '
+                    'Valid values for `allowed_states` and `failed_states` '
                     'when `external_task_id` is not `None`: {}'.format(State.task_states)
                 )
         else:
-            if not set(self.allowed_states) <= set(State.dag_states):
+            if not set(total_states) <= set(State.dag_states):
                 raise ValueError(
-                    'Valid values for `allowed_states` '
+                    'Valid values for `allowed_states` and `failed_states` '
                     'when `external_task_id` is `None`: {}'.format(State.dag_states)
                 )
```

Comment:
`error_template = 'Valid values for `allowed_states` and `failed_states` '\
                 'when `external_task_id` is not `None`: {}'
if external_task_id and not set(total_states) <= set(State.task_states):
    raise ValueError(error_template.format(State.task_states))
elif not set(total_states) <= set(State.dag_states):
    raise ValueError(error_template.format(State.dag_states))`
Jus my personal view :P